### PR TITLE
Remove CollegeHumor from embed providers

### LINKED ISSUE #18567: Block Library: CollegeHumor should be removed
Originally reported on the Gutenberg plugin support forum at https://wordpress.org/support/topic/collegehumor-should-be-removed/ by [slkfsdf8y34ljhsfsdfkuhfkl84hj](https://wordpress.org/support/users/slkfsdf8y34ljhsfsdfkuhfkl84hj/):

> CollegeHumor block should be removed as the site is no longer an independant website, they now redirect it to YouTube.
> 
> I always wondered why it was a block anyway, there are so much more popular websites comparing to that one that were not added.

### PR DESCRIPTION
**Pull request by audrasjb**
CollegeHumor doesn't support oEmbed anymore, so this PR removes the related blocks.

For reference, see the related Trac ticket: https://core.trac.wordpress.org/ticket/48696
Fixes #18567

Cheers,
Jb

### CODE REVIEW

**Review by gziolo — CHANGES_REQUESTED** [review]
Thanks @audrasjb for starting this PR. I'm afraid we can't simply remove this embed because this will create backward compatibility issues for posts created in the past. Users would see the message like:
`Your site doesn’t include support for the "core-embed/collegehumor" block...`

There is some prior art though where @ice9js added a deprecation logic for the Crowdsignal embed. See #12854. I think it can be replicated here, the difference is that we don't need to add a new embed but probably change the pattern for the YouTube embed.

**Comment by audrasjb** [comment]
Thanks for the review @gziolo !
I updated the patch accordingly :)

File: packages/block-library/src/embed/core-embeds.js
Comment (gziolo) [inline packages/block-library/src/embed/core-embeds.js]: I would expect this array to become empty from now on to avoid matching those patterns with this particular embed. When you paste the link in the empty Paragraph block, it will probably transform automatically. I'm not sure what will happen with YouTube links in this case as well.

**Comment by mcsf** [comment]
> I'm afraid we can't simply remove this embed because this will create backward compatibility issues for posts created in the past. […]
> 
> There is some prior art though […]

I came here because of the same thing. This is definitely something we need to fix in the embed blocks framework. Embed blocks probably just need to be multiple expressions of a single block type, so that we don't need to maintain these things _ad hoc_.

**Comment by audrasjb** [comment]
Ok! Thanks @gziolo, makes sense.
I updated the patch accordingly.

**Comment by gziolo** [comment]
I googled to find some old url, example:
http://www.collegehumor.com/video/60123495/real-adults-dont-order-dessert

Screen Shot 2019-11-21 at 12 59 45


When I embed it, it get recognized as a general embed. Is that okay?

**Comment by audrasjb** [comment]
@gziolo Looks good to me.
FYI, non embed will be generated at all outside the block editor.
https://core.trac.wordpress.org/attachment/ticket/48696/Capture%20d%E2%80%99%C3%A9cran%202019-11-18%20%C3%A0%2020.47.52.png

**Review by gziolo — APPROVED** [review]
We will have to re-test it once the core patch lands.

Let's move forward as is. I applied the label `Needs Dev Note` to ensure we revisit it later. I assume we will need a note for WordPress 5.4 release for the core part anyway.

**Comment by gziolo** [comment]
One more thing to unblock merging, https://travis-ci.com/WordPress/gutenberg/jobs/258989943#L608:

```
 /home/travis/build/WordPress/gutenberg/packages/block-library/src/embed/core-embeds.js
609[0]   148:5  error  Missing trailing comma  comma-dangle
610[0] 
611[0] ✖ 1 problem (1 error, 0 warnings)
612[0]   1 error and 0 warnings potentially fixable with the `--fix` option.
```

`npm run lint-js:fix` should be all that is necessary.

**Comment by mcsf** [comment]
Following up on https://github.com/WordPress/gutenberg/pull/18591#issuecomment-557045134, although this doesn't address the issue of a single block type expressed as many blocks, it does "work" as a quick quick concept of handling missing embeds:

```diff
diff --git a/packages/blocks/src/api/parser.js b/packages/blocks/src/api/parser.js
index 6300eac32..4182375a9 100644
--- a/packages/blocks/src/api/parser.js
+++ b/packages/blocks/src/api/parser.js
@@ -455,9 +455,17 @@ export function createBlockWithFallback( blockNode ) {
 			innerHTML = originalContent;
 		}
 
-		name = unregisteredFallbackBlock;
-		attributes = { originalName, originalContent, originalUndelimitedContent };
-		blockType = getBlockType( name );
+		// If detected as an embed whose definition is missing, automatically
+		// convert to generic embed.
+		if ( name.indexOf( 'core-embed/' ) === 0 ) {
+			name = 'core/embed';
+			blockType = getBlockType( name );
+			innerHTML = originalUndelimitedContent;
+		} else {
+			name = unregisteredFallbackBlock;
+			attributes = { originalName, originalContent, originalUndelimitedContent };
+			blockType = getBlockType( name );
+		}
 	}
 
 	// Coerce inner blocks from parsed form to canonical form.
```

**Comment by gziolo** [comment]
@mcsf - did you just propose a way to deprecate all individual embeds and integrate them as one block type with different provides defined as patterns? 😃

**Comment by mcsf** [comment]
😆

**Comment by audrasjb** [comment]
@gziolo I tried `npm run lint-js:fix` but merging is still blocked, sorry 😬

**Comment by mcsf** [comment]
There seem to be some intermittent test failures. Re-running failed tests.

File: package-lock.json
Comment (gziolo) [inline package-lock.json]: There are some unrelated changes included in this file. I bet they should be removed from this patch.

Comment (audrasjb) [reply package-lock.json]: This is the result of `npm run lint-js:fix` command on my branch. Should I removed them with a new commit?

Comment (gziolo) [reply package-lock.json]: It's rather from `npm install` :)
Yes, please remove it.

Comment (audrasjb) [reply package-lock.json]: Thanks @gziolo, we should be good to go now :)

**Comment by gziolo** [comment]
Thanks @audrasjb for this patch 👍

**Comment by audrasjb** [comment]
For your information, the related Core ticket has been committed and reopened for 5.3.1 consideration.
https://core.trac.wordpress.org/ticket/48696

**Pull request merged**
